assessment/collectors/infoschema_collector_test.go

```diff
@@ -0,0 +1,923 @@
+package assessment
+
+import (
+	"database/sql"
+	"fmt"
+	"testing"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/assessment/sources/mysql"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/assessment/utils"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/common/constants"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/internal"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/logger"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/profiles"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/schema"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/spanner/ddl"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+)
+
+type MockInfoSchema struct {
+	tables           map[string]utils.TableAssessmentInfo
+	indexes          []utils.IndexAssessmentInfo
+	triggers         []utils.TriggerAssessmentInfo
+	storedProcedures []utils.StoredProcedureAssessmentInfo
+	functions        []utils.FunctionAssessmentInfo
+	views            []utils.ViewAssessmentInfo
+	err              error
+}
+
+func init() {
+	logger.Log = zap.NewNop()
+}
+
+type MockConnectionConfigProvider struct {
+	config interface{}
+	err    error
+}
+
+func (m MockConnectionConfigProvider) GetConnectionConfig(sourceProfile profiles.SourceProfile) (interface{}, error) {
+	return m.config, m.err
+}
+
+type MockDBConnector struct {
+	db  *sql.DB
+	err error
+}
+
+func (m MockDBConnector) Connect(driver string, connectionConfig interface{}) (*sql.DB, error) {
+	return m.db, m.err
+}
+
+func TestCreateInfoSchemaCollector(t *testing.T) {
+	db, mock, err := sqlmock.New()
+	if err != nil {
+		t.Fatalf("an error '%s' was not expected when opening a stub database connection", err)
+	}
+	defer db.Close()
+
+	dbName := "test_db"
+	tableName := "table1"
+	columnName := "column1"
+	indexName := "index1"
+	triggerName := "trigger1"
+	procedureName := "procedure1"
+	functionName := "function1"
+	viewName := "view1"
+
+	mock.ExpectQuery(`SELECT TABLE_COLLATION, SUBSTRING_INDEX\(TABLE_COLLATION, '_', 1\) as CHARACTER_SET\s+FROM INFORMATION_SCHEMA.TABLES\s+WHERE TABLE_SCHEMA = \? AND TABLE_NAME = \?`).
```

Comment:
It might help to move the specific features into methods and then have tests for each, rather than having a large test mocking so much stuff.